### ISSUE TITLE
List comprehensions

### ISSUE THREAD

**Issue by Misterion777** [apache/incubator-age#105]
Hey, 
are list comprehensions supported in AGE?
Using syntax from Neo4j doesnt seem to work, simple example:
```
SELECT * FROM cypher('g1', $$
MATCH (v)
RETURN [x in v.array_field]
$$) as (a agtype);
```

**Comment by Halkrine** [comment]
You mean that this query(array_field) is working or not with AGE?

I tried that query. Once it works as 'query' (return as jsonb from Pgadmin). 

But I'm not sure this result is correct(Because it returns null). 


Someone who knows this issue?

**Comment by Misterion777** [comment]
> You mean that this query(array_field) is working or not with AGE?
> 
> I tried that query. Once it works as 'query' (return as jsonb from Pgadmin).
> 
> But I'm not sure this result is correct(Because it returns null).
> 
> Someone who knows this issue?

array_field is some random array which is an attribute of a graph vertex.
i'm trying it with AGE and receiving following error:
```
ERROR:  variable `x` does not exist
LINE 3: RETURN [x in v.test_ids]
```

**Comment by Halkrine** [comment]
replace x -> v and try it.

**Comment by Misterion777** [comment]
> replace x -> v and try it.

no, it should be another variable: it should iterate through the array like in any list comprehension

**Comment by Halkrine** [comment]
Understood.

Maybe, that function is not supported yet now..

**Comment by JoshInnis** [comment]
Hi, List Comprehensions are not currently supported

**Comment by jasperblues** [comment]
Its a useful feature. Not sure if it will be the case on AGE but with Neo4j they can especially help with performance.

**Comment by jrgemignani** [comment]
List comp. was "nearly" completed until it was found to have serious issues for any nested cases. It will be revisited soon.
